google/cloud/bigquery/v2/minimal/internal/bigquery_http_response.cc

```diff
@@ -42,6 +45,19 @@ StatusOr<BigQueryHttpResponse> BigQueryHttpResponse::BuildFromRestResponse(
   return response;
 }
 
+std::string BigQueryHttpResponse::DebugString(
```

Comment:
We've added support for a `DebugFormatter` in "google/cloud/internal/debug_string.h" to help create consistently-formatted strings for request/response logging (consistent with the proto messages).

I updated the `DebugString()` functions for [`GetJobRequest`](https://github.com/googleapis/google-cloud-cpp/blob/main/google/cloud/bigquery/v2/minimal/internal/job_request.cc#L78-L86) and [`ListJobsRequest`](https://github.com/googleapis/google-cloud-cpp/blob/main/google/cloud/bigquery/v2/minimal/internal/job_request.cc#L88-L110) to use `DebugFormatter`.  Perhaps you could try it here.

You may have issues with repeated-ish fields (like `BigQueryHttpResponse.http_headers`), or sub-messages (like `GetJobResponse.http_response` and `ListJobsResponse.http_response`).  If so, it would be nice to understand them so that we can improve `DebugFormatter`.

Reply:
> You may have issues with ... sub-messages

I added `DebugFormatter` support for sub-messages, but it means that any `DebugString()` functions using this framework will now also take a leading "name" parameter, which will be used to produce the `name { ... }` part.

The intent is that the caller from the decorator will pass the full type name, while `SubMessage()` callers will pass the field name.

> You may have issues with repeated-ish fields

I'll look into that today.

None of this, of course, means you have to adopt `DebugFormatter` here.  We can address later.